File: nowcoder/构建乘积数组.py
# encoding: utf-8

'''

@author: liangchi

@contact: bnu_llc@163.com

@software: pycharm

@file: 构建乘积数组.py

@time: 2018/5/29 下午3:08

'''


def multiply( A):
    # write code here
    res, temp = [1], 1
    for i in A[:-1]:
        temp *= i
        res.append(temp)
    print res

    temp = 1
    for index in range(len(A) - 1, 0, -1):
        temp *= A[index]
        res[index-1] *= temp
    return res


if __name__ == '__main__':
    print multiply( [1, 2, 3, 4, 5])

File: sword_c++/构建乘积数组.cpp
//
// Created by 李先生 on 2018/9/4.
//

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> multiply(const vector<int>& A) {
        vector<int> res;
        int temp = 1;
        res.push_back(temp);
        for(int i = 1 ;i < A.size(); i++)
        {
            temp *= A[i-1];
            res.push_back(temp);
        }
        temp = 1;
        for (int i= A.size()-1; i > 0; i--)
        {
            temp *= A[i];
            res[i-1] *= temp;
        }
        return res;
    }
};

int main()
{
    Solution s;
    vector<int> a = {1,2,3,4,5};
    vector <int> res = s.multiply(a);
    for(int i = 0; i<res.size(); i++)
        cout<<res[i]<<endl;
    return  0;
}